lib/dal/src/secret.rs

```diff
@@ -291,6 +318,65 @@ impl Secret {
         self.key
     }
 
+    /// Attach a [`Secret`] to a given [`AttributeValue`] corresponding to a
+    /// "/root/secrets/\<secret\>" [`Prop`](crate::Prop).
+    ///
+    /// - If a [`Secret`] has already been attached to the given [`AttributeValue`], the existing
+    ///   attachment will be replaced with a new one.
+    /// - If no [`Secret`] is provided, then the [`AttributeValue`] will be updated to its original
+    ///   state via [`AttributeValue::use_default_prototype`].
+    ///
+    /// This method will enqueue
+    /// [`DependentValuesUpdate`](crate::job::definition::DependentValuesUpdate).
+    pub async fn attach_for_attribute_value(
+        ctx: &DalContext,
+        attribute_value_id: AttributeValueId,
+        secret_id: Option<SecretId>,
+    ) -> SecretResult<()> {
+        // First, check if the caller would like to unset the secret.
+        let secret_id = match secret_id {
+            Some(provided_secret_id) => provided_secret_id,
+            None => {
+                AttributeValue::use_default_prototype(ctx, attribute_value_id).await?;
```

Comment:
We use the default prototype here rather than passing `None` to `AttributeValue::update` because we neither want nor need existing prototype arguments and the user cannot override the prototype for values corresponding to `/root/secrets/<secret>`.